contracts/pallet-ibc/src/ics20/mod.rs

```diff
@@ -561,11 +563,370 @@ pub fn full_ibc_denom(packet: &Packet, mut token: PrefixedCoin) -> String {
+						from: receiver.clone(),
+						to: account_to.clone(),
+						para_id: memo_forward.para_id,
+						amount,
+						asset_id: asset_id.into(),
+					});
+					Ics20Error::implementation_specific(
+						"Faield to execute SubstrateMultihopXcmHandler::transfer_xcm.".to_string(),
+					)
+				})?;
+
+				crate::Pallet::<T>::deposit_event(Event::<T>::ExecuteMemoXcmSuccess {
+					from: receiver.clone(),
+					to: account_to.clone(),
+					para_id: memo_forward.para_id,
+					amount,
+					asset_id: asset_id.into(),
+				});
+
+				return Ok(())
+			},
+		};
+
+		// let raw_bytes = memo_forward.receiver.as_bytes().to_vec();
+		let raw_bytes = memo_forward.receiver.into_bytes();
+
+		//convert string into T::AccountId
+
+		let transfer_ibc_account_to =
+			crate::MultiAddress::<<T as frame_system::Config>::AccountId>::Raw(raw_bytes.clone());
+
+		let origin = RawOrigin::Signed(receiver.clone());
+		let channel_id = memo_forward
+			.channel
+			.split('-')
+			.last()
+			.ok_or_else(|| {
+				Self::emit_memo_execution_failed_event(
+					receiver.clone(),
+					packet_data.memo.clone(),
+					3,
+				);
+				Ics20Error::implementation_specific(format!(
+					"Failed to extract channel number from channel ID: {:?}",
+					memo_forward.channel
+				))
+			})?
+			.parse()
+			.map_err(|_| {
+				Self::emit_memo_execution_failed_event(
+					receiver.clone(),
+					packet_data.memo.clone(),
+					4,
+				);
+				Ics20Error::implementation_specific("Failed to parse channel ID".to_string())
+			})?;
+
+		//timestamp that current timestamp + 30 minutes
+		let _timestamp = <pallet_timestamp::Pallet<T> as frame_support::traits::UnixTime>::now()
+			.as_secs() + 1800;
+
+		let params = crate::TransferParams::<<T as frame_system::Config>::AccountId> {
+			to: transfer_ibc_account_to,
+			source_channel: channel_id,
+			// timeout: ibc_primitives::Timeout::Offset { timestamp: Some(timestamp), height: None
+			// }, timestamp: Some(600), height: Some(600) is working 100% so it is better to test
+			// this first before to complicate
+			timeout: ibc_primitives::Timeout::Offset { timestamp: Some(600), height: Some(600) },
+		};
+
+		let memo = memo.forward.next.as_deref();
+
+		let mut next_memo: Option<T::MemoMessage> = None;
+		if let Some(memo) = memo {
+			let memo_str = serde_json::to_string(&memo).map_err(|_| {
+				Self::emit_memo_execution_failed_event(
+					receiver.clone(),
+					packet_data.memo.clone(),
+					5,
+				);
+				Ics20Error::implementation_specific("failed to serialize memo".to_string())
+			})?;
+			let memo_result =
+				<T as crate::Config>::MemoMessage::from_str(&memo_str).map_err(|_| {
+					Self::emit_memo_execution_failed_event(
+						receiver.clone(),
+						packet_data.memo.clone(),
+						6,
+					);
+					Ics20Error::implementation_specific(
+						"failed to convert string to Config::MemoMessage".to_string(),
+					)
+				})?;
+			next_memo = Some(memo_result);
+		}
+
+		crate::Pallet::<T>::transfer(
```

Comment:
I remember having some issues with middleware due to the packets not being removed from storage properly. Could this be a case here in case of failure @vmarkushin ?

Reply:
No, should be good